packages/cardano-services/src/WsServer/db.ts

```diff
@@ -0,0 +1,68 @@
+// cSpell:ignore plpgsql proname prosrc tgfoid tgname
+
+import { Cardano } from '@cardano-sdk/core';
+import { Logger } from 'ts-log';
+import { Pool } from 'pg';
+
+const notifyTipBody = `\
+BEGIN
+  PERFORM pg_notify('sdk_tip', json_build_object(
+    'blockNo', NEW.block_no,
+    'hash',    encode(NEW.hash, 'hex'),
+    'slot',    NEW.slot_no
+  )::TEXT);
+  RETURN NEW;
+END;`;
+
+const notifyTipProcedure = `\
+CREATE OR REPLACE FUNCTION sdk_notify_tip()
+RETURNS TRIGGER AS
+$BODY$${notifyTipBody}$BODY$
+LANGUAGE plpgsql`;
+
+const notifyTipTrigger = `\
+CREATE TRIGGER sdk_notify_tip
+AFTER INSERT ON block
+FOR EACH ROW EXECUTE PROCEDURE sdk_notify_tip();`;
+
+export const initDB = async (db: Pool, logger: Logger) => {
+  logger.info('Initializing the DB...');
+
+  // Wait for a block .i.e the schema is ready
+  let blocks: number;
+  do {
+    logger.debug('Checking for latest block...');
+    const { rows } = await db.query<{ blocks: number }>('SELECT COUNT(*) AS blocks FROM block');
+    blocks = rows[0].blocks;
+  } while (blocks === 0);
```

Comment:
For my own understanding, why do we want to have blocks before handling the triggers?

Reply:
That's a way to wait for db-sync schema to be ready.
It's just a check to be sure the table is created before attaching a trigger to it.